ospd/ospd.py

```diff
@@ -1045,18 +1047,25 @@ def handle_get_vts_command(self, vt_et):
             return simple_response_str('get_vts', 404, text)
 
         filtered_vts = None
-        if vt_filter:
+        if not vt_id and vt_filter:
             filtered_vts = self.vts_filter.get_filtered_vts_list(
                 self.vts, vt_filter
             )
+        elif vt_id:
+            filtered_vts = vt_id
+        else:
+            filtered_vts = self.vts.keys()
 
-        responses = []
-
-        vts_xml = self.get_vts_xml(vt_id, filtered_vts)
+        yield xml_helper.create_response('get_vts')
+        yield xml_helper.create_element('vts')
 
-        responses.append(vts_xml)
+        for vt in self.get_vt_generator():
```

Comment:
Personally I would call this iterator because a generator is a special (the most common type) iterator.
```suggestion
        for vt in self.get_vt_iterator():
```

Reply:
Done. Also, I implemented the iterator in the class, in case it was not implemented in the wrapper, and it still can get the VTs in case there is a vts dictionary already loaded.